Client/Application/Delegates/AppDelegate.swift

```diff
@@ -260,9 +260,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
         }
         
         if let urp = UserReferralProgram.shared {
-            if isFirstLaunch {
+            if Preferences.URP.referralLookupOutstanding.value == nil {
+                // This preference has never been set, and this means it is a new or upgraded user.
+                // That distinction must be made to know if a network request for ref-code look up should be made.
+                
+                // Setting this to an explicit value so it will never get overwritten on subsequent launches.
+                // Upgrade users should not have ref code ping happening.
+                Preferences.URP.referralLookupOutstanding.value = isFirstLaunch
+            }
+            
+            if Preferences.URP.referralLookupOutstanding.value == true {
                 let refCode = UserReferralProgram.sanitize(input: UIPasteboard.general.string)
-                if refCode != nil { UIPasteboard.general.clearPasteboard() }
+                if refCode != nil {
+                    UrpLog.log("Clipboard ref code found: " + (UIPasteboard.general.string ?? "!Clipboard Empty!"))
+                    UrpLog.log("Clearing clipboard.")
+                    UIPasteboard.general.clearPasteboard()
```

Comment:
If the refcode is cleared here, app won't find the refcode on the next launch if the the 10 network requests are not successful. May be store it in URP preferences?

Reply:
Ref code is passed into the look up function, and it is stored internally there:
https://github.com/brave/brave-ios/pull/2574/files#diff-bfe2070e3d8f02013654953e18b60730R280

Reply:
I think my concern is that we are still accessing the clipboard on subsequent launches, I think we should just store it in a preference and use it from there: https://github.com/brave/brave-ios/pull/2574/files#diff-bfe2070e3d8f02013654953e18b60730R273 - Let me know if I'm missing something.

Reply:
That make sense, we will add a ref code check to prevent reading the clipboard again. Will be in a follow PR though, as it doesn't pertain to what this PR is.